File: Interesting Patterns/Interesting_Patterns.java
class Solution {
    static String[] interestingPattern(int n) {
        String[] ans = new String[2 * n - 1];
        int index = 0;
        
        for (int i = n; i > 0; i--) {
            StringBuilder temp = new StringBuilder();
            for (int j = n; j > 0; j--) {
                if (j >= i) temp.append(j);
                else temp.append(i);
            }
            String tt = temp.toString();
            temp.reverse();
            String result = tt.substring(0, tt.length() - 1) + temp.toString();
            ans[index++] = result;
        }
        
        for (int i = 2; i <= n; i++) {
            StringBuilder temp = new StringBuilder();
            for (int j = n; j > 0; j--) {
                if (j >= i) temp.append(j);
                else temp.append(i);
            }
            String tt = temp.toString();
            temp.reverse();
            String result = tt.substring(0, tt.length() - 1) + temp.toString();
            ans[index++] = result;
        }
        
        return ans;
    }
};

File: Interesting Patterns/Interesting_Patterns.py
class Solution:
    def interestingPattern(self, n):
        ans = []
        
        for i in range(n, 0, -1):
            temp = ""
            for j in range(n, 0, -1):
                if j >= i:
                    temp += str(j)
                else:
                    temp += str(i)
            tt = temp
            temp = tt[:-1] + temp[::-1]
            ans.append(temp)
        
        for i in range(2, n + 1):
            temp = ""
            for j in range(n, 0, -1):
                if j >= i:
                    temp += str(j)
                else:
                    temp += str(i)
            tt = temp
            temp = tt[:-1] + temp[::-1]
            ans.append(temp)
        
        s = "";
        for i in ans:
            s += i+"\n"
        return s    
